src/librustc_driver/lib.rs

```diff
@@ -640,23 +612,20 @@ impl RustcDefaultCalls {
                     let allow_unstable_cfg = UnstableFeatures::from_environment()
                         .is_nightly_build();
 
-                    for cfg in &sess.parse_sess.config {
-                        if !allow_unstable_cfg && GatedCfg::gate(cfg).is_some() {
+                    for &(name, ref value) in sess.parse_sess.config.iter() {
```

Comment:
Hmm, doesn't this expose `HashSet` ordering? Even if you were to use a `BTreeSet` it would expose intern order. Maybe collect strings instead of printing them, to a `Vec` and just sort that?